### ISSUE TITLE
Prompt_Tuning.ipynb example doesn't seem to train the model

### ISSUE THREAD

**Issue by ruixing76** [huggingface/peft#2642]
I am running Prompt-Tuning notebook example from PEFT lib examples [here](https://github.com/huggingface/peft/blob/main/examples/sequence_classification/Prompt_Tuning.ipynb).  I did **not** change any line of code and I ran the code block sequentially.

However, the performance under metrics remain exactly the **same** for each epoch, which is very weird. From the [orignal notebook](https://github.com/huggingface/peft/blob/main/examples/sequence_classification/Prompt_Tuning.ipynb), we can see accuracy fluctuates and can increase to 0.70. 

I checked the output logits for the training data is changing every epoch (set shuffle=False, and this is the only change for debugging). Now I am very confused, any suggestions would be very much welcome, please let me know if I am doing something very wrong, thanks in advance!

Here's the performance log:
```
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.36it/s]
epoch 0: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.72it/s]
100%|██████████| 13/13 [00:01<00:00, 10.49it/s]
epoch 1: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.34it/s]
epoch 2: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.72it/s]
100%|██████████| 13/13 [00:01<00:00, 10.35it/s]
epoch 3: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.47it/s]
epoch 4: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.69it/s]
100%|██████████| 13/13 [00:01<00:00, 10.63it/s]
epoch 5: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.75it/s]
100%|██████████| 13/13 [00:01<00:00, 10.45it/s]
epoch 6: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.40it/s]
epoch 7: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.53it/s]
epoch 8: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:19<00:00,  5.76it/s]
100%|██████████| 13/13 [00:01<00:00, 10.27it/s]
epoch 9: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.75it/s]
100%|██████████| 13/13 [00:01<00:00, 10.50it/s]
epoch 10: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.63it/s]
epoch 11: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:19<00:00,  5.77it/s]
100%|██████████| 13/13 [00:01<00:00, 10.50it/s]
epoch 12: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:19<00:00,  5.78it/s]
100%|██████████| 13/13 [00:01<00:00, 10.60it/s]
epoch 13: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
100%|██████████| 115/115 [00:20<00:00,  5.74it/s]
100%|██████████| 13/13 [00:01<00:00, 10.54it/s]
epoch 14: {'accuracy': 0.6838235294117647, 'f1': 0.8122270742358079}
```

Besides, my environment info is here if it helps debugging:
```
python  3.10
transformers  4.52.4
peft  0.16.0
torch 2.7.0
jupyterlab 4.4.3
OS Ubunt 22.04 LTS
GPU NVIDIA RTX 5880
```

**Comment by BenjaminBossan** [comment]
What you observe is indeed strange and I can't reproduce it on my end. ~For some reasons, your scores already start at a high value, as if the model was already trained at this point. I'm really unsure how that can be if you run the exact same notebook. Just for testing, could you please vary the learning rate and see if it changes anything?~ **Update**: I think what is happening is that the model already converged after the first epoch. Try changing the learning rate (probably lowering it) and see if that helps.

While investigating this, I did, however, find a bug in PEFT when it comes to prompt tuning and sequence classification. Could you please try using PEFT v0.14 and see if it resolves the issue? I'll work on a fix in the meantime.

**Comment by ruixing76** [comment]
> What you observe is indeed strange and I can't reproduce it on my end. ~For some reasons, your scores already start at a high value, as if the model was already trained at this point. I'm really unsure how that can be if you run the exact same notebook. Just for testing, could you please vary the learning rate and see if it changes anything?~ **Update**: I think what is happening is that the model already converged after the first epoch. Try changing the learning rate (probably lowering it) and see if that helps.
> 
> While investigating this, I did, however, find a bug in PEFT when it comes to prompt tuning and sequence classification. Could you please try using PEFT v0.14 and see if it resolves the issue? I'll work on a fix in the meantime.

Thanks @BenjaminBossan, I can reproduced this on other nodes with the same GPU type. I tried the following solultions:
- For lr, I changed it from 1e-3 to 1e-4 and it produced the same, and changed it to 1e-5 will generate 0 f1 all the time (probabaly it's too small).
- After I switched to PEFT v0.14, the metric computation seems to be normal and converges fast. Now I use 0.14 for now.
```
100%|██████████| 115/115 [00:19<00:00,  5.90it/s]
100%|██████████| 13/13 [00:01<00:00, 10.55it/s]
epoch 0: {'accuracy': 0.6887254901960784, 'f1': 0.812960235640648}
100%|██████████| 115/115 [00:19<00:00,  5.76it/s]
100%|██████████| 13/13 [00:01<00:00, 10.24it/s]
epoch 1: {'accuracy': 0.6985294117647058, 'f1': 0.8172362555720654}
100%|██████████| 115/115 [00:20<00:00,  5.64it/s]
100%|██████████| 13/13 [00:01<00:00, 10.18it/s]
epoch 2: {'accuracy': 0.7083333333333334, 'f1': 0.8221225710014948}
100%|██████████| 115/115 [00:20<00:00,  5.57it/s]
100%|██████████| 13/13 [00:01<00:00, 10.18it/s]
epoch 3: {'accuracy': 0.7107843137254902, 'f1': 0.8233532934131736}
100%|██████████| 115/115 [00:20<00:00,  5.59it/s]
100%|██████████| 13/13 [00:01<00:00, 10.28it/s]
epoch 4: {'accuracy': 0.7303921568627451, 'f1': 0.8318042813455657}
100%|██████████| 115/115 [00:20<00:00,  5.62it/s]
100%|██████████| 13/13 [00:01<00:00, 10.18it/s]
epoch 5: {'accuracy': 0.7156862745098039, 'f1': 0.8258258258258259}
```

Another thing is the convergence for PEFT in prompt-tuning varies on different GPU types, on NVIDIA RTX 5880 it seems faster (with all hyperparameters the same, fix batch order and random seeds). I knew there might be some backend architecture stuff but first epoch performance exhibit the difference of 5% on F1.

If possible, can you tell me the bug you identified? Thanks!

**Comment by BenjaminBossan** [comment]
> If possible, can you tell me the bug you identified? Thanks!

There was a regression that resulted in the classification layer not being trained (it is supposed to be fully fine-tuned, no PEFT) in the 0.15.0 release. This is fixed in #2646. However, I'm not sure if it fully explains your observation. You could install PEFT from main and see if the issue is fixed for you.

> convergence for PEFT in prompt-tuning varies on different GPU types, on NVIDIA RTX 5880 it seems faster (with all hyperparameters the same, fix batch order and random seeds). I knew there might be some backend architecture stuff but first epoch performance exhibit the difference of 5% on F1.

It's hard to say in the abstract, what other hardware are you using? Is the dtype the same for the two?

**Comment by ruixing76** [comment]
> > If possible, can you tell me the bug you identified? Thanks!
> 
> There was a regression that resulted in the classification layer not being trained (it is supposed to be fully fine-tuned, no PEFT) in the 0.15.0 release. This is fixed in [#2646](https://github.com/huggingface/peft/pull/2646). However, I'm not sure if it fully explains your observation. You could install PEFT from main and see if the issue is fixed for you.
> 
> > convergence for PEFT in prompt-tuning varies on different GPU types, on NVIDIA RTX 5880 it seems faster (with all hyperparameters the same, fix batch order and random seeds). I knew there might be some backend architecture stuff but first epoch performance exhibit the difference of 5% on F1.
> 
> It's hard to say in the abstract, what other hardware are you using? Is the dtype the same for the two?

Thanks, I only have these 2 types of GPU. I swtiched to 0.14 and it seems to perform normally like in the last comment.

**Comment by BenjaminBossan** [comment]
>  I only have these 2 types of GPU.

One is an NVIDIA RTX 5880, what is the other?

> You could install PEFT from main and see if the issue is fixed for you.

Did you have the opportunity to test this and run training on those 2 GPUs?

**Comment by ruixing76** [comment]
> > I only have these 2 types of GPU.
> 
> One is an NVIDIA RTX 5880, what is the other?
> 
> > You could install PEFT from main and see if the issue is fixed for you.
> 
> Did you have the opportunity to test this and run training on those 2 GPUs?

Oops, sorry I forgot to mention! The other one is RTX 6000 Ada, the performance looks more normal.

**Comment by BenjaminBossan** [comment]
> Oops, sorry I forgot to mention! The other one is RTX 6000 Ada, the performance looks more normal.

Hmm, both GPUs use the same Ada Lovelace architecture, so they should behave the same. Could there be differences in driver or CUDA versions?

> > You could install PEFT from main and see if the issue is fixed for you.
> 
> Did you have the opportunity to test this and run training on those 2 GPUs?

It would be great if you could test this. Install PEFT from main using `https://github.com/huggingface/peft` and then re-run the training using both GPUs. Please report the full logs for both.
